lib/SPIRV/OCLToSPIRV.cpp

```diff
@@ -70,6 +70,23 @@ static size_t getOCLCpp11AtomicMaxNumOps(StringRef Name) {
       .Default(0);
 }
 
+static Type *getBlockStructType(Value *Parameter) {
+  // In principle, this information should be passed to us from Clang via
+  // an elementtype attribute. However, said attribute requires that the
+  // function call be an intrinsic, which it is not. Instead, we rely on being
+  // able to trace this to the declaration of a variable: OpenCL C specification
+  // section 6.12.5 should guarantee that we can do this.
+  Value *UnderlyingObject = getUnderlyingObject(Parameter);
```

Comment:
`getUnderlyingObject` has default `MaxLookup` parameter of 6. Does it make sense to explicitly pass 0 to make sure, that we have no limitations for number of the instructions to strip off?

Reply:
I copied `getUnderlyingObject` from the original version of this code, but now that I think about it, `stripPointerCasts` is probably the better function anyways--it will look through any number of 0-offset geps and bitcasts, but will stop if there's a non-zero offset on the process.